### ISSUE TITLE
Invalid Connection

### ISSUE THREAD

**Issue by Jhutchings5** [Azure/LogicAppsUX#4226]
### Describe the Bug with repro steps 

When you look at the UI for a successful workflow run, you get the 'green check' indicated that the action was successful, however, after a few seconds (roughly 5-10 seconds) it will display that there is an invalid connection.

### What type of Logic App Is this happening in? 

Standard (Portal)

### Are you using new designer or old designer 

New Designer

### Did you refer to the TSG before filing this issue? [https://aka.ms/lauxtsg](<https://aka.ms/lauxtsg>) 

Yes

### Workflow JSON 

```json
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {
            "Upload_blob_to_storage_container": {
                "inputs": {
                    "parameters": {
                        "blobName": "@triggerOutputs()['headers']['x-ms-file-name']",
                        "containerName": "icaudit",
                        "content": "@triggerBody()"
                    },
                    "serviceProviderConfiguration": {
                        "connectionName": "AzureBlob",
                        "operationId": "uploadBlob",
                        "serviceProviderId": "/serviceProviders/AzureBlob"
                    }
                },
                "runAfter": {},
                "type": "ServiceProvider"
            }
        },
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "triggers": {
            "When_a_file_is_added_or_modified": {
                "inputs": {
                    "host": {
                        "connection": {
                            "referenceName": "sftpwithssh"
                        }
                    },
                    "method": "get",
                    "path": "/datasets/default/triggers/onupdatedfile",
                    "queries": {
                        "folderId": "L2FkaG9jLXRyYW5zZmVycw==",
                        "includeFileContent": true,
                        "inferContentType": true,
                        "queryParametersSingleEncoded": true
                    }
                },
                "metadata": {
                    "6e202211-2856-4d17-9ded-5beb8b8626b0": "/",
                    "L2FkaG9jLXRyYW5zZmVycw==": "/adhoc-transfers"
                },
                "recurrence": {
                    "frequency": "Hour",
                    "interval": 1,
                    "startTime": "2024-02-04T06:00:00Z",
                    "timeZone": "Eastern Standard Time"
                },
                "type": "ApiConnection"
            }
        }
    },
    "kind": "Stateful"
}
```

### Screenshots or Videos

![error](<https://github.com/Azure/LogicAppsUX/assets/39990692/1db0bcad-652d-41c2-8964-3ccae7faa7b2>)

### Browser

Edge

### Additional context

The fallback ui doesn't seem to have this issue. This only seems to exist on the new UI side

**Comment by Jhutchings5** [comment]
This might be expected.  The SFTP connection is a managed api connection, the only permissions on it are the standard logic app system assigned identity.

**Comment by rllyy97** [comment]
@jhutchings5  Just to be clear on the issue I have a few questions, is the connection expected to be valid currently?  Or was it valid during the run and now it's invalid? (ie. we don't care that the connection is invalid in the present time)

Mainly I'm wondering if this is an issue with the connection improperly being shown as invalid, or just that we don't need to see the invalid connection warnings in the monitoring view

**Comment by Jhutchings5** [comment]
> @Jhutchings5 Just to be clear on the issue I have a few questions, is the connection expected to be valid currently? Or was it valid during the run and now it's invalid? (ie. we don't care that the connection is invalid in the present time)
> 
> Mainly I'm wondering if this is an issue with the connection improperly being shown as invalid, or just that we don't need to see the invalid connection warnings in the monitoring view

Good Morning,

I am expecting the connection to be valid.  There doesn't seem to be any issues with the connectivity at all.  When I create the action and attach the connection, everything seems fine.  I am able to see the SFTP connection and view the folders to where it is pointing.  Also, the run seems to work fine.

**Comment by rllyy97** [comment]
@Jhutchings5  I'm investigating another issue I think might have the same root cause, it happens when you try to create connections to both Built-In and Azure versions of a connector in the same logic app.  If you're trying to connect to the Built-In version do you also see a reference to another SFTP connector in the connections section of your logic app?

If you are okay doing so you could send me a network trace of you loading the app and I'd be able to see if there's any sort of connection overlap, you can find info on how to grab these logs from our docs here: https://purple-pebble-0e11de71e.1.azurestaticapps.net/tsg#collecting-console-errors
And my email is just RileyEvans@microsoft.com

**Comment by Jhutchings5** [comment]
Good Morning Riley,

It appears that you mail provider is rejecting me sending you the HAR file (its too large).

Is there another way that I can provide you the data?

Thank you,

Josh Hutchings

From: Riley Evans ***@***.***>
Sent: Thursday, February 29, 2024 5:22 PM
To: Azure/LogicAppsUX ***@***.***>
Cc: Joshua Hutchings ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [Azure/LogicAppsUX] Invalid Connection (Issue #4226)


@Jhutchings5 [github.com]<https://urldefense.com/v3/__https:/github.com/Jhutchings5__;!!Hk2OhvHRu-6GlA!Q35ejkxswArIWRpeDFzVX0e-nTRjp8HOzsgJsM7GXPeRIIsaoKw830NJmwaz4JGchzpby4tUm0CL75XQgXlfOHPCBzMM$> I'm investigating another issue I think might have the same root cause, it happens when you try to create connections to both Built-In and Azure versions of a connector in the same logic app. If you're trying to connect to the Built-In version do you also see a reference to another SFTP connector in the connections section of your logic app?

If you are okay doing so you could send me a network trace of you loading the app and I'd be able to see if there's any sort of connection overlap, you can find info on how to grab these logs from our docs here: https://urldefense.com/v3/__https://purple-pebble-0e11de71e.1.azurestaticapps.net/tsg*collecting-console-errors__;Iw!!Hk2OhvHRu-6GlA!T0ENDVl2xNZVK4-wVf45mbx6VxBAzD78rpL2JXRBBvqoVBh1dJYGMALmEszE32CV8NjI_MamEXA2HpjKaP7OLL5jinLfpZUZfqWnURltG2tSL9bPGhnnbifEZcOaD2P9L-EWBFpCIqTL$  [purple-pebble-0e11de71e.1.azurestaticapps.net]<https://urldefense.com/v3/__https:/purple-pebble-0e11de71e.1.azurestaticapps.net/tsg*collecting-console-errors__;Iw!!Hk2OhvHRu-6GlA!Q35ejkxswArIWRpeDFzVX0e-nTRjp8HOzsgJsM7GXPeRIIsaoKw830NJmwaz4JGchzpby4tUm0CL75XQgXlfOBudsCAG$>
And my email is just ***@***.******@***.***>

**Comment by rllyy97** [comment]
@Jhutchings5  One thing you could try is opening the workflow run in this staging environment: https://ms.portal.azure.com/?Microsoft_Azure_EMA=staging#home
The staging environment currently has a fix that removes some unnecessary requests in the monitoring view, so it might be small enough to send if you do it there.
You could also send just your connections file, found here, then JSON view
![image](https://github.com/Azure/LogicAppsUX/assets/25409734/6e28ee37-dbe5-4d6f-9ae3-c1a53b8b57d3)

All of the secrets and such should already be stored as appsettings but if you see any keys you're concerned about feel free to remove them

**Comment by Jhutchings5** [comment]
Here is the connections JSON. I also sent the smaller HAR file via email.

`{
    "managedApiConnections": {
        "sftpwithssh": {
            "api": {
                "id": "/subscriptions/9fa37a05-fd9b-40ec-99f4-505874d3508a/providers/Microsoft.Web/locations/eastus/managedApis/sftpwithssh"
            },
            "authentication": {
                "type": "ManagedServiceIdentity"
            },
            "connection": {
                "id": "/subscriptions/9fa37a05-fd9b-40ec-99f4-505874d3508a/resourceGroups/@appsetting('WORKFLOWS_RESOURCE_GROUP_NAME')/providers/Microsoft.Web/connections/@appsetting('instacartSftp_name')"
            },
            "connectionRuntimeUrl": "@appsetting('SFTP_CONNECTION_RUNTIMEURL')"
        }
    },
    "serviceProviderConnections": {
        "AzureBlob": {
            "displayName": "securityBlobStorage",
            "parameterSetName": "connectionString",
            "parameterValues": {
                "connectionString": "@appsetting('AzureBlob_blobStorageEndpoint')"
            },
            "serviceProvider": {
                "id": "/serviceProviders/AzureBlob"
            }
        }
    }
}`

**Comment by Jhutchings5** [comment]
So I actually figured out the issue.

I am not sure as to why this happened, however, the managedApiConnection sftwithssh was created in a different subscription than the others. (its a different subscription for our prod environment).  I am guessing that this value was being set statically in the pipeline and was setting the wrong value.  Once I changed the value to the correct subscription.  It worked just fine and that error message went away.

I apologize for not catching this sooner.
